### PR TITLE
Unbreak Circle CI, and other fixes

### PR DESCRIPTION
**Pull request by philmd**
Fix the Circle CI:
- Update Fedora kernel URL
- Use recent Fedora with recent QEMU
- Few fixes to avoid the script looping for hours on CI

Due to Tianocore BZ#1747 change in EnrollDefaultKeys,
generate a temporary X509 certificate to run the test.

### CODE REVIEW

**Review by kashyapc — COMMENTED** [review]
Fedora-29 is dead, please use Fedora-31.

Thanks for figuring out the way to un-break Circle CI,

**Review by kashyapc — DISMISSED** [review]

File: ovmf-vars-generator
Comment (kashyapc) [inline ovmf-vars-generator]: Ah-ha, makes sense; looks good to me.

Reviewed-by: Kashyap Chamarthy <kchamart@redhat.com>

File: .circleci/Dockerfile
Comment (kashyapc) [inline .circleci/Dockerfile:1]: So the "latest", didn't work out as we expected (as the image never got refreshed).

Explicitly spelling out the version is fine.

File: .circleci/config.yml
Comment (kashyapc) [inline .circleci/config.yml]: Thanks for fixing this temporarily, Phil.  Okay, we can change this back once Patrick comes and updates his Docker repo.

**Review by kashyapc — DISMISSED** [review]

**Comment by kashyapc** [comment]
Thanks, Phil.  Now that Laszlo's patch is merged; we can take it take this a bit less hurried :-)  It looks good to me, I'll let Laszlo / Patrick also have a further look at this.

File: .circleci/config.yml
Comment (philmd) [reply .circleci/config.yml]: Indeed, now that #28 is merged we can wait until Patrick upgrades his DockerHub image first, and simply drop this patch before merging the series.

**Review by lersek — DISMISSED** [review]
Reviewed-by: Laszlo Ersek <lersek@redhat.com>

(to clarify, I meant to attach my R-b / approval specifically to patch " Generate temporary X509 certificate to be enrolled as Platform Key". I'm very unfamiliar with the github.com patch review / commenting system)

**Review by lersek — DISMISSED** [review]
In patch *Generate temporary X509 certificate to be enrolled as Platform Key*, passing the `--oem-string ...` option should not be necessary in the *run testing-only* case (i.e. when `--skip-enrollment` is passed). Namely, if we don't enroll, then we don't have to pass down the certificate to enroll.

But that's a minor wart only; I'm happy with this series going in. If Phil chooses to drop the one `--oem-string ...` instance in question, then my R-b can be carried forward. For the series ending at commit 805210979832:

`Reviewed-by: Laszlo Ersek <lersek@redhat.com>`

Thanks!

**Comment by philmd** [comment]
> In patch _Generate temporary X509 certificate to be enrolled as Platform Key_, passing the `--oem-string ...` option should not be necessary in the _run testing-only_ case (i.e. when `--skip-enrollment` is passed). Namely, if we don't enroll, then we don't have to pass down the certificate to enroll.
> 
> But that's a minor wart only; I'm happy with this series going in. If Phil chooses to drop the one `--oem-string ...` instance in question, then my R-b can be carried forward. For the series ending at commit [8052109](https://github.com/puiterwijk/qemu-ovmf-secureboot/commit/805210979832c06bb4567c966ad0fe6495081659):

Thanks for the review, I did the requested change (Kashyap approving and yours got reset).

**Review by lersek — APPROVED** [review]
Based on

`git range-diff 78f00c0511a5  80521097983 79be1ee7a247`

I only need to provide feedback on commit 48d819cf50fb ("Generate temporary X509 certificate to be enrolled as Platform Key", 2020-01-02). My R-b, existent on that patch, stands. Thanks!

File: .circleci/Dockerfile
Comment (kashyapc) [inline .circleci/Dockerfile:6]: F27 has reached End-of-Life; here too, please use Fedora 31.

**Review by kashyapc — APPROVED** [review]
Looks fine to me; I see that the DockerHub is now changed to 'rhuefi' (https://hub.docker.com/u/rhuefi)

**Pull request merged**
